### PR TITLE
Bump Swift Argument Parser checkout to 1.1.4

### PR DESCRIPTION
**Pull request by daltonclaybrook**
This changes the checked-out version of swift-argument-parser from 1.0.3 to 1.1.2 in `update-checkout-config.json` as requested in [apple/swift-driver#1106](https://github.com/apple/swift-driver/pull/1106) . Please let me know if any other changes are required as I'm not very familiar with how this file is used. I took my cues from #41126

### CODE REVIEW

File: utils/update_checkout/update-checkout-config.json [hunk 1]
```diff
@@ -177,7 +177,7 @@
                 "llbuild": "release/5.7",
                 "swift-tools-support-core": "release/5.7",
                 "swiftpm": "release/5.7",
-                "swift-argument-parser": "1.0.3",
+                "swift-argument-parser": "1.1.2",
```
Comment (compnerd) [inline utils/update_checkout/update-checkout-config.json:180]: Please revert this part of the change - if this is intended to go into 5.7, let us first settle main and then look at 5.7 (though I suspect that it is a bit too late for that)

**Comment by compnerd** [comment]
@swift-ci please build toolchain macOS platform

**Comment by compnerd** [comment]
@swift-ci please build toolchain Linux platform

**Comment by compnerd** [comment]
Please test with following PRs:
apple/swift-installer-script#91

@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
Please test with following PRs:
apple/swift-installer-scripts#91

@swift-ci please build toolchain Windows platform

File: utils/update_checkout/update-checkout-config.json (hunk 1 above)
Comment (daltonclaybrook) [reply utils/update_checkout/update-checkout-config.json]: @compnerd ✅

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/swift-installer-scripts/pull/91

@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/swift-installer-scripts/pull/91

@swift-ci please build toolchain Linux platform

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/swift-installer-scripts/pull/91

@swift-ci please build toolchain macOS platform

**Comment by compnerd** [comment]
@swift-ci please clean test

**Comment by CodaFi** [comment]
https://github.com/apple/swift-installer-scripts/pull/91

@swift-ci please build toolchain macOS platform

**Comment by CodaFi** [comment]
https://github.com/apple/swift-installer-scripts/pull/91

@swift-ci please build toolchain Windows platform

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-installer-scripts/pull/91

@swift-ci please build toolchain Windows platform

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

File: utils/update_checkout/update-checkout-config.json [hunk 2]
```diff
@@ -93,7 +93,7 @@
                 "llbuild": "main",
                 "swift-tools-support-core": "main",
                 "swiftpm": "main",
-                "swift-argument-parser": "1.0.3",
+                "swift-argument-parser": "1.1.2",
```
Comment (MaxDesiatov) [inline utils/update_checkout/update-checkout-config.json:96]: Would you mind updating this to 1.1.4, as that version had a few nice improvements on top?

Comment (daltonclaybrook) [reply utils/update_checkout/update-checkout-config.json]: Done

File: utils/update_checkout/update-checkout-config.json [hunk 3]
```diff
@@ -135,7 +135,7 @@
...
```
Comment (MaxDesiatov) [inline utils/update_checkout/update-checkout-config.json:138]: The other two places where the version is mentioned need to be updated as well.

Comment (daltonclaybrook) [reply utils/update_checkout/update-checkout-config.json]: Done 👍

Comment (MaxDesiatov) [reply utils/update_checkout/update-checkout-config.json]: Thanks!

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci please clean test

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test with following PRs
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218

**Comment by MaxDesiatov** [comment]
@swift-ci please clean test
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please clean test

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please clean test

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please clean test

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please build toolchain

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please clean test

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please build toolchain

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please clean test

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please build toolchain

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please clean test macOS

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please build macOS toolchain

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please build toolchain

**Comment by MaxDesiatov** [comment]
@compnerd Windows toolchain build failure seems unrelated and is apparently caused by 
```
The following tests FAILED:
	606 - TestFoundation.TestFileManager-test_homedirectoryForUser (ILLEGAL)
	611 - TestFoundation.TestFileManager-test_emptyFilename (Exit code 0xc0000409
)
	1281 - TestFoundation.TestNSString-test_NSHomeDirectoryForUser (Failed)
	1283 - TestFoundation.TestNSString-test_expandingTildeInPath (Failed)
```
have you seen this before? AFAIU the Foundation repository doesn't have Windows CI setup to reproduce this test failure separately

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please clean test macOS

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci build macOS toolchain

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please build macOS toolchain

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please build toolchain

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please clean test

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please build toolchain

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please clean test Linux

**Comment by MaxDesiatov** [comment]
@swift-ci please test Windows

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please clean test Windows

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please build toolchain Windows platform

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please build toolchain Windows platform

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please build toolchain Windows platform

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please build toolchain Windows platform

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please clean test

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
@swift-ci please build toolchain

**Comment by MaxDesiatov** [comment]
@compnerd could you confirm that the Windows failure is not related to this PR? Thanks!

**Comment by compnerd** [comment]
@MaxDesiatov can't do that - we need to get a complete toolchain build for Windows because there can be a lurking issue - swift-installer-scripts may need to be updated.

**Comment by compnerd** [comment]
Please test with following PRs:
apple/sourcekit-lsp#689

@swift-ci please build toolchain Windows platfrom

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
https://github.com/apple/sourcekit-lsp/pull/689
@swift-ci please build toolchain Windows platform

**Review by compnerd — APPROVED** [review]
I _think_ that this may need https://github.com/apple/swift-installer-scripts/pull/91 to be merged in tandem to ensure that we don't break the Windows build.

**Comment by compnerd** [comment]
@MaxDesiatov - when this goes in, we need to make a change to compnerd/swift-build as well to ensure that the official Windows builds are updated as well.

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
https://github.com/apple/sourcekit-lsp/pull/689
@swift-ci please build toolchain Windows platform

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
https://github.com/apple/sourcekit-lsp/pull/689
@swift-ci please build toolchain Windows platform

**Comment by MaxDesiatov** [comment]
@compnerd it failed with
```
Archiving artifacts
‘build/artifacts/*’ doesn’t match anything, but ‘*’ does. Perhaps that’s what you mean?
ERROR: Step ‘Archive the artifacts’ failed: No artifacts found that match the file pattern "build/artifacts/*". Configuration error?
```
what would be the best way to diagnose this?

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
https://github.com/apple/sourcekit-lsp/pull/689
@swift-ci please build toolchain Windows platform

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
https://github.com/apple/sourcekit-lsp/pull/673
@swift-ci please build toolchain Windows platform

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
https://github.com/apple/sourcekit-lsp/pull/673
@swift-ci clean test

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
https://github.com/apple/sourcekit-lsp/pull/673
@swift-ci clean test

**Comment by MaxDesiatov** [comment]
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
https://github.com/apple/sourcekit-lsp/pull/673
@swift-ci build toolchain

**Comment by compnerd** [comment]
@MaxDesiatov the build logs are the most helpful thing here:
```
swift-ci@EC2AMAZ-JCVOGM7 C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows>msbuild C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows\swift-installer-scripts\platforms\Windows\toolchain.wixproj   -p:RunWixToolsOutOfProc=true   -p:OutputPath=T:\package\toolchain\   -p:IntermediateOutputPath=T:\package\toolchain\   -p:DEVTOOLS_ROOT=T:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain   -p:TOOLCHAIN_ROOT=T:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain 
Microsoft (R) Build Engine version 16.10.2+857e5a733 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 1/5/2023 3:14:53 PM.
Project "C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows\swift-installer-scripts\platforms\Windows\toolchain.wixproj" on node 1 (default targets).
PrepareForBuild:
  Creating directory "T:\package\toolchain\".
HarvestDirectory:
  C:\Program Files (x86)\WiX Toolset v3.11\bin\Heat.exe dir T:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\lib\clang -cg ClangResources -dr _usr_lib_clang -scom -sreg -srd -var var.TOOLCHAIN_ROOT_USR_LIB_CLANG -nologo -gg -nologo -sfrag -out T:\package\toolchain\_ClangResources_dir.wxs
Compile:
  C:\Program Files (x86)\WiX Toolset v3.11\bin\candle.exe -dProductVersion=0.0.0 -dDEVTOOLS_ROOT=T:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain -dTOOLCHAIN_ROOT=T:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain -dTOOLCHAIN_ROOT_USR_LIB_CLANG=T:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\lib\clang -dConfiguration=Debug -dOutDir=T:\package\toolchain\ -dPlatform=AnyCPU -dProjectDir=C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows\swift-installer-scripts\platforms\Windows\ -dProjectExt=.wixproj -dProjectFileName=toolchain.wixproj -dProjectName=toolchain -dProjectPath=C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows\swift-installer-scripts\platforms\Windows\toolchain.wixproj -dTargetDir=T:\package\toolchain\ -dTargetExt=.msi -dTargetFileName=toolchain.msi -dTargetName=toolchain -dTargetPath=T:\package\toolchain\toolchain.msi -out T:\package\toolchain\ -ext "C:\Program Files (x86)\WiX Toolset v3.11\bin\\WixUIExtension.dll" toolchain-amd64.wxs T:\package\toolchain\_ClangResources_dir.wxs
Link:
  C:\Program Files (x86)\WiX Toolset v3.11\bin\Light.exe -out T:\package\toolchain\toolchain.msi -pdbout T:\package\toolchain\toolchain.wixpdb -cultures:null -ext "C:\Program Files (x86)\WiX Toolset v3.11\bin\\WixUIExtension.dll" -contentsfile T:\package\toolchain\toolchain.wixproj.BindContentsFileListnull.txt -outputsfile T:\package\toolchain\toolchain.wixproj.BindOutputsFileListnull.txt -builtoutputsfile T:\package\toolchain\toolchain.wixproj.BindBuiltOutputsFileListnull.txt -wixprojectfile C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows\swift-installer-scripts\platforms\Windows\toolchain.wixproj T:\package\toolchain\toolchain-amd64.wixobj T:\package\toolchain\_ClangResources_dir.wixobj
C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows\swift-installer-scripts\platforms\Windows\toolchain-amd64.wxs(685): error LGHT0103: The system cannot find the file 'T:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\bin\ArgumentParserToolInfo.dll'. [C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows\swift-installer-scripts\platforms\Windows\toolchain.wixproj]
Done Building Project "C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows\swift-installer-scripts\platforms\Windows\toolchain.wixproj" (default targets) -- FAILED.

Build FAILED.

"C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows\swift-installer-scripts\platforms\Windows\toolchain.wixproj" (default target) (1) ->
(Link target) -> 
  C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows\swift-installer-scripts\platforms\Windows\toolchain-amd64.wxs(685): error LGHT0103: The system cannot find the file 'T:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\bin\ArgumentParserToolInfo.dll'. [C:\Users\swift-ci\jenkins\workspace\swift-PR-build-toolchain-windows\swift-installer-scripts\platforms\Windows\toolchain.wixproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:05.20
```
https://github.com/apple/swift-installer-scripts/pull/91 is what is needed.

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/swift-installer-scripts/pull/91
https://github.com/apple/swift-package-manager/pull/5884
https://github.com/apple/swift-driver/pull/1218
https://github.com/apple/sourcekit-lsp/pull/673

@swift-ci please build toolchain windows platform

**Comment by MaxDesiatov** [comment]
@compnerd could it also be Foundation test failures that have some impact?

```
The following tests FAILED:
	606 - TestFoundation.TestFileManager-test_homedirectoryForUser (ILLEGAL)
	611 - TestFoundation.TestFileManager-test_emptyFilename (Exit code 0xc0000409
)
	1281 - TestFoundation.TestNSString-test_NSHomeDirectoryForUser (Failed)
	1283 - TestFoundation.TestNSString-test_expandingTildeInPath (Failed)
Errors while running CTest
```

**Comment by compnerd** [comment]
The Foundation tests are something that I am hoping to look into soon, but the error is what I referenced in https://github.com/apple/swift/pull/59009#issuecomment-1372383082.  It results in a missing MSI as the manifest is stating that a file needs to be shipped but is not available due to changes in swift-argument-parser.  The result is that we cannot build the MSI nor installer as the manifest is no longer correct.

**Review by shahmishal — APPROVED** [review]

**Pull request merged**